resource-management/pkg/common-lib/metrics/event_metrics.go

```diff
@@ -0,0 +1,130 @@
+
+func init() {
+	latencyMetricsAllCheckpoints = new(LatencyMetricsAllCheckpoints)
+	latencyMetricsAllCheckpoints.Aggregator_Received = NewLatencyMetrics(int(Aggregator_Received))
+	latencyMetricsAllCheckpoints.Distributor_Received = NewLatencyMetrics(int(Distributor_Received))
+	latencyMetricsAllCheckpoints.Distributor_Sending = NewLatencyMetrics(int(Distributor_Sending))
+	latencyMetricsAllCheckpoints.Distributor_Sent = NewLatencyMetrics(int(Distributor_Sent))
+	latencyMetricsAllCheckpoints.Serializer_Encoded = NewLatencyMetrics(int(Serializer_Encoded))
+	latencyMetricsAllCheckpoints.Serializer_Sent = NewLatencyMetrics(int(Serializer_Sent))
+}
+
+func AddLatencyMetricsAllCheckpoints(e runtime.Object) {
+	if !common_lib.ResourceManagementMeasurement_Enabled {
+		return
+	}
+	if e == nil {
+		klog.Error("Nil event")
+	}
+	checkpointsPerEvent := e.GetCheckpoints()
+	if checkpointsPerEvent == nil {
+		klog.Errorf("Event (%v, Id %s, RV %v) does not have checkpoint stamped", e.GetEventType(), e.GetId(), e.GetResourceVersionInt64())
+	}
+	lastUpdatedTime := e.GetLastUpdatedTime()
+
+	agg_received_time := checkpointsPerEvent[Aggregator_Received]
+	dis_received_time := checkpointsPerEvent[Distributor_Received]
+	dis_sending_time := checkpointsPerEvent[Distributor_Sending]
+	dis_sent_time := checkpointsPerEvent[Distributor_Sent]
+	serializer_encoded_time := checkpointsPerEvent[Serializer_Encoded]
+	serializer_sent_time := checkpointsPerEvent[Serializer_Sent]
+
+	errMsg := fmt.Sprintf("Event (%v, Id %s, RV %v)", e.GetEventType(), e.GetId(), e.GetResourceVersionInt64()) + " does not have %s stamped"
+	latencyMetricsLock.Lock()
+	defer latencyMetricsLock.Unlock()
+	if !agg_received_time.IsZero() {
+		latencyMetricsAllCheckpoints.Aggregator_Received.AddLatencyMetrics(agg_received_time.Sub(lastUpdatedTime))
+	} else {
+		klog.Errorf(errMsg, Aggregator_Received)
+	}
+	if !dis_received_time.IsZero() {
+		latencyMetricsAllCheckpoints.Distributor_Received.AddLatencyMetrics(dis_received_time.Sub(lastUpdatedTime))
+	} else {
+		klog.Errorf(errMsg, Distributor_Received)
+	}
+	if !dis_sending_time.IsZero() {
+		latencyMetricsAllCheckpoints.Distributor_Sending.AddLatencyMetrics(dis_sending_time.Sub(lastUpdatedTime))
+	} else {
+		klog.Errorf(errMsg, Distributor_Sending)
+	}
+	if !dis_sent_time.IsZero() {
+		latencyMetricsAllCheckpoints.Distributor_Sent.AddLatencyMetrics(dis_sent_time.Sub(lastUpdatedTime))
+	} else {
+		klog.Errorf(errMsg, Distributor_Sent)
+	}
+	if !serializer_encoded_time.IsZero() {
+		latencyMetricsAllCheckpoints.Serializer_Encoded.AddLatencyMetrics(serializer_encoded_time.Sub(lastUpdatedTime))
+	} else {
+		klog.Errorf(errMsg, Serializer_Encoded)
+	}
+	if !serializer_sent_time.IsZero() {
+		latencyMetricsAllCheckpoints.Serializer_Sent.AddLatencyMetrics(serializer_sent_time.Sub(lastUpdatedTime))
+	} else {
+		klog.Errorf(errMsg, Serializer_Sent)
+	}
+	klog.V(6).Infof("[Metrics][Detail] node %v RV %v: %s: %v, %s: %v, %s: %v, %s: %v, %s: %v, %s: %v",
```

Comment:
the defer unlock code in line 72 can be moved to before the klog code.

Reply:
> the defer unlock code in line 72 can be moved to before the klog code.

Moved and removed defer